pkg/util/http/transport.go

```diff
@@ -126,8 +126,12 @@ func GetProxyTransportFunc(p *config.Proxy) func(*http.Request) (*url.URL, error
 		}(r)
 
 		// Print a warning if the proxy behavior would change if the new no_proxy behavior would be enabled
-		newURL, _ := proxyConfig.ProxyFunc()(r.URL)
-		if url != newURL && url != nil {
+		var newURLString string
+		if newURL, _ := proxyConfig.ProxyFunc()(r.URL); newURL != nil {
+			newURLString = newURL.String()
+		}
+
+		if url != nil && url.String() != newURLString {
```

Comment:
This condition doesn't perfectly match the meaning of the warning that's logged:

- here the condition tests that the 2 resolved proxies' string URLs are exactly the same, and not nil
- whereas the message in the warning log indicates that we've detected a narrower state: that `url` is not nil and `newURLString` _is nil_

If there are known cases where we expect the results of the 2 proxy resolution logics to yield non-nil but different proxies for a given URL, we should investigate them (and maybe not log them as warnings, but at another level) so we can make this case part of the migration plan.

Thinking about this a bit more: for a given URL, the proxy resolution logic should only return one of 3 values:
- `nil`
- the proxy configured for HTTP requests (under `HTTP_PROXY`, `DD_PROXY_HTTP`, or `proxy.http`)
- the proxy configured for HTTPS requests (under `HTTPS_PROXY`, `DD_PROXY_HTTPS`, or `proxy.https`)

And reading the code of `httpproxy.ProxyFunc` (https://golang.org/src/vendor/golang.org/x/net/http/httpproxy/proxy.go?s=4158:4228#L108), I can't find an obvious difference in the logic that could lead to different results here. Might still be worth logging something if the results are different, but both values aren't `nil`?